pelican/settings.py

```diff
@@ -509,7 +509,19 @@ def configure_settings(settings):
 
     # specify the log messages to be ignored
     log_filter = settings.get('LOG_FILTER', DEFAULT_CONFIG['LOG_FILTER'])
-    LimitFilter._ignore.update(set(log_filter))
+    for item in log_filter:
+        if len(item) == 2:  # old-style string or template
+            LimitFilter.ignore.update({item})
+        elif len(item) == 3:  # new-style string/template or regexp
+            if item[0] == "string":
+                LimitFilter.ignore.update({(item[1:])})
+            elif item[0] == "regex":
+                regex = re.compile(item[2])
+                LimitFilter.ignore_regexp.update({(item[1], regex)})
+            else:
+                raise ValueError(f"Invalid LOG_FILTER type '{item[0]}'")
+        else:
+            raise ValueError(f"Invalid item '{str(item)}' in LOG_FILTER")
```

Comment:
I think it makes more sense to move this logic to `LimitFilter` instead, say a `@classmethod` named `add_ignore_rule`.

Also, `LimitFilter.ignore.update({item})` is a long way of writing `LimitFilter.ignore.add(item)` :).